pkg/oss/utils.go

```diff
@@ -145,3 +145,29 @@ func setTransmissionProtocol(originURL string) (URL string, modified bool) {
 	}
 	return "https://" + URL, true
 }
+
+// translateDomainToEndpoint will trim the prefix with "bucket." to
+//
+//	translate the bucket domain name like: bucket.oss-cn-beijing.aliyuncs.com
+//	to endpoint like: oss-cn-beijing.aliyuncs.com
+func translateDomainToEndpoint(originURL, bucket string) (URL string, modified bool) {
```

Comment:
We have many URL manipulation now. How about using URL parsing (https://pkg.go.dev/net/url#Parse) instead of string manipulation?

Reply:
Existing URL manipulations seem optimized. Correcting wrong URLs isn't our responsibility, I believe. besides, if we did, maybe this is more correcting things in the future.

Reply:
We have similar discussion before, the point is, the 'url' here is OSS endpoint in fact... not a real url, and OSS has so many deprecated endpoints.